spring-auto-restdocs-core/src/main/java/capital/scalable/restdocs/request/RequestParametersSnippet.java

```diff
@@ -46,9 +46,20 @@ public RequestParametersSnippet failOnUndocumentedParams(boolean failOnUndocumen
 
     @Override
     protected boolean isRequired(MethodParameter param, RequestParam annot) {
-        return param.getParameterType().isPrimitive()
-                ? ValueConstants.DEFAULT_NONE.equals(annot.defaultValue())
-                : annot.required();
+        if (param.getParameterType().isPrimitive()) {
+            // a primitive type is required if no defaultValue is set
+            return !hasDefaultValue(annot);
+        } else {
+            if (hasDefaultValue(annot)) {
+                // Having a defaultValue set implies required=false
+                return false;
+            }
+            return annot.required();
+        }
```

Comment:
I would suggest to restructure the conditions to make clear that the default value in all cases implies that it is not required and also to avoid checking the condition twice, for example:
```
        if (hasDefaultValue(annot)) {
            // Having a defaultValue set implies required=false
            return false;
         } else if (param.getParameterType().isPrimitive()) {
             // A primitive type is required if no defaultValue is set, regardless of the value of the required flag
             return true;
         } else {
             return annot.required();
         }
```

Reply:
That makes sense. I applied this change in 5e09b0c7a37f732795eeb2e8618967a27f44fb4f.